src/Snippets/Docs/ResilienceContextUsage.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Polly;
+
+namespace Snippets.Docs;
+
+internal static class ResilienceContextUsage
+{
+    public static async Task Usage()
+    {
+        var cancellationToken = CancellationToken.None;
+
+        #region resilience-context
+
+        // Retrieve a context with a cancellation token
+        ResilienceContext context = ResilienceContextPool.Shared.Get(cancellationToken);
+
+        // Attach custom data to the context
+        var key1 = new ResiliencePropertyKey<string>("my-key-1");
+        var key2 = new ResiliencePropertyKey<int>("my-key-2");
+
+        context.Properties.Set(key1, "my-data");
+        context.Properties.Set(key2, 123);
+
+        // Utilize the context in a resilience pipeline
+        ResiliencePipeline pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new()
+            {
+                OnRetry = args =>
+                {
+                    // Retrieve custom data from the context, if available
+                    if (args.Context.Properties.TryGetValue(key1, out var data))
+                    {
+                        Console.WriteLine("OnRetry, Custom Data: {0}", data);
+                    }
+
+                    return default;
+                }
+            })
+            .Build();
```

Comment:
Should we maybe write this sample in a way that doesn't allocate the closure to get the keys to pull out the data?

If not, maybe maybe make a note about that somewhere for advanced scenarios?